redfish-exporter/redfish_utils.go

```diff
@@ -189,6 +190,45 @@ func deleteSubscriptionFromServer(server RedfishServer, subscriptionURI string)
 	return nil
 }
 
+// Unsubscribes/deletes overlapping subscriptions from the server
+func DeleteExistingSubscription(server RedfishServer, subscriptionPayload SubscriptionPayload) error {
+	subscriptions, err := GetServerSubscriptions(server)
+	if err != nil {
+		return err
+	}
+	for _, subscription := range subscriptions {
+		if subscription.Destination == subscriptionPayload.Destination {
+			err := deleteSubscriptionFromServer(server, subscription.ODataID)
+			if err != nil {
+				return fmt.Errorf("failed to delete event subscription %s, on server %s: %v", subscription.ID, server.IP, err)
+			} else {
+				log.Printf("successfully deleted overlapping event subscription %s from server %s", subscription.ID, server.IP)
+			}
+		}
+	}
+	return nil
+}
+
+// Gets all subscriptions currently active on the given server
+func GetServerSubscriptions(server RedfishServer) ([]*redfish.EventDestination, error) {
+	c, err := getRedfishClient(server)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to redfish server %s: %v", server.IP, err)
+	}
+	defer c.Logout()
+
+	es, err := c.Service.EventService()
```

Comment:
Also, one good practice is to use the same name across the project. so either eventService or es.

Reply:
And, just my 2 cents, longer names that are more human readable please (as mentioned, but in case others read this). Compiler makes this shorter when run, autocomplete makes it easier to type, so longer, more descriptive names that humans will look at are always a better option ;)

Reply:
I agree 100% 😄 Go's naming guide recommends short names when the variable is used close by (apparently easier to scan?), but I prefer descriptive names myself :)